test/summary.cc

```diff
@@ -0,0 +1,153 @@
+              "max_route_distance_factor":5,"max_route_time_factor":5,"max_search_radius":200,"route":true,
+              "search_radius":15.0,"sigma_z":4.07,"turn_penalty_factor":200}},
+      "service_limits": {
+        "auto": {"max_distance": 5000000.0, "max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "auto_shorter": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "bicycle": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50},
+        "bus": {"max_distance": 5000000.0,"max_locations": 50,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "hov": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "isochrone": {"max_contours": 4,"max_distance": 25000.0,"max_locations": 1,"max_time": 120},
+        "max_avoid_locations": 50,"max_radius": 200,"max_reachability": 100,"max_alternates":2,
+        "multimodal": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 0.0,"max_matrix_locations": 0},
+        "pedestrian": {"max_distance": 250000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50,"max_transit_walking_distance": 10000,"min_transit_walking_distance": 1},
+        "skadi": {"max_shape": 750000,"min_resample": 10.0},
+        "trace": {"max_distance": 200000.0,"max_gps_accuracy": 100.0,"max_search_radius": 100,"max_shape": 16000,"max_best_paths":4,"max_best_paths_shape":100},
+        "transit": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50},
+        "truck": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50}
+      }
+    })";
+  boost::property_tree::ptree conf;
+  rapidjson::read_json(ss, conf);
+  return conf;
+}
+
+struct route_tester {
+  route_tester()
+      : conf(get_conf()), reader(new GraphReader(conf.get_child("mjolnir"))),
+        loki_worker(conf, reader), thor_worker(conf, reader), odin_worker(conf) {
+  }
+  Api test(const std::string& request_json) {
+    Api request;
+    ParseApi(request_json, Options::route, request);
+    loki_worker.route(request);
+    thor_worker.route(request);
+    odin_worker.narrate(request);
+    loki_worker.cleanup();
+    thor_worker.cleanup();
+    odin_worker.cleanup();
+    return request;
+  }
+  boost::property_tree::ptree conf;
+  std::shared_ptr<GraphReader> reader;
+  loki_worker_t loki_worker;
+  thor_worker_t thor_worker;
+  odin_worker_t odin_worker;
+};
+
+TEST(Summary, test_time_summary) {
+  route_tester tester;
+  std::string request =
+      R"({"locations":[{"lat":52.048267,"lon":5.074825},{"lat":52.114622,"lon":5.131816}],"costing":"auto"})";
+  auto response = tester.test(request);
+
+  // loop over all routes all legs
```

Comment:
first part of the test is to loop over the internal protobuf route representations, both the path and the directions objects and compare that they all add up to the same thing